ResidentInformationApi/V1/Gateways/AcademyInformationGateWay.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Boundary.Responses;
+
+namespace ResidentInformationApi.V1.Gateways
+{
+    public class AcademyInformationGateway : IAcademyInformationGateway
+    {
+        private readonly HttpClient _client;
+        private readonly string _baseUrl;
+
+        public AcademyInformationGateway(HttpClient client)
+        {
+            _client = client;
+            _baseUrl = Environment.GetEnvironmentVariable("ACADEMY_API_ENDPOINT");
+        }
+        public async Task<List<AcademyClaimantInformation>> GetClaimantInformation(ResidentQueryParam rqp)
+        {
+            var queryDictionary = new Dictionary<string, string>();
+            if (!string.IsNullOrEmpty(rqp.FirstName)) queryDictionary.Add("first_name", rqp.FirstName);
+            if (!string.IsNullOrEmpty(rqp.LastName)) queryDictionary.Add("last_name", rqp.LastName);
+            if (!string.IsNullOrEmpty(rqp.Address)) queryDictionary.Add("address", rqp.Address);
+            if (!string.IsNullOrEmpty(rqp.Postcode)) queryDictionary.Add("postcode", rqp.Postcode);
+            var rqpString = new FormUrlEncodedContent(queryDictionary).ReadAsStringAsync().Result;
```

Comment:
It might be nice to refactor this into a query parameter -> query string method in some sort of helper method class so that it can be reused in all of the gateways, as it will be used in all of them.

Reply:
Good shout, I've extracted it. Though this is the first time I've extracted a method in C# before, so I'd appreciate pointers on improvement!